tests/infrastructure/vm_console_proxy/utils.py

```diff
@@ -66,3 +70,27 @@ def get_vm_console_proxy_resource(resource_kind, namespace=None):
             name=f"{TOKEN_API_VERSION}.{TOKEN_ENDPOINT}" if resource_kind == APIService else VM_CONSOLE_PROXY
         )
     return vm_console_proxy_resource_object
+
+
+def assert_resource_status_and_wait_for_availability(resource: Any, timeout: int) -> None:
+    assert resource.exists, (
+        f"Missing: {resource.kind}/{resource.name} under {getattr(resource, 'namespace', 'cluster')}"
+    )
+
```

Comment:
This should not be part of this function. If you are checking status and waiting for conditions, the resource should exist at that point.